packages/nx/src/command-line/targets.ts

```diff
@@ -0,0 +1,35 @@
+import { output } from '../utils/output';
+import { createProjectGraphAsync } from '../project-graph/project-graph';
+import { ProjectGraph } from '../config/project-graph';
+import { ProjectConfiguration } from '../config/workspace-json-project-json';
+
+/**
+ * Lists available targets for a workspace
+ *
+ * @remarks
+ *
+ * Must be run within an Nx workspace
+ *
+ */
+export async function targetsHandler(): Promise<void> {
+  const projectGraph: ProjectGraph<ProjectConfiguration> =
+    await createProjectGraphAsync({ exitOnError: true });
+
+  const bodyLines: string[] = [];
+  for (const [project, { data }] of Object.entries(projectGraph.nodes)) {
```

Comment:
This will be A LOT of different targets when there are a lot of projects.

I don't think it will be helpful to print all of these.